static/apps/style/style-app.js

```diff
@@ -33,7 +33,7 @@ define([
             // kicks off any objects and processes that need to run
```

Comment:
regarding the "layerHasBeenAltered" and "layerHasBeenSaved" flags: I think that there's a Backbone model property called model.hasChanged() that can be used to check (versus manually tracking this).